### PR TITLE
[TF FE] Fix bad cast for tf.TensorShape to ov.PartialShape

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Fix bad cast for tf.TensorShape to ov.PartialShape. The error occurred in mobile-object-localizer-v1 model.

**Ticket:** CVS-132455

### CODE REVIEW

File: src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py [hunk 1]
```diff
@@ -26,25 +26,33 @@ def tf_type_to_ov_type(tf_type_int):
...
-    if attr_type == "list":
+    elif attr_type == "list":
         list_value = attr.list
         fields = list_value.ListFields()
         if fields and len(fields) > 0 and len(fields[0]) > 1:
             return list(fields[0][1])
         else:
             return None
-    if attr_type is None:
+    elif attr_type == "shape":
```
Comment (popovaan) [inline src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py:48]: Can we delete the branch checking `name == "shape" or name == "_output_shapes"` (lines 147-158)?

Comment (rkazants) [reply src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py:48]: it is better to leave such refactoring for the next release and to have it not risky, now I just cover remained cases.

Comment (rkazants) [reply src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py:48]: Also, added check if variable is `None`

File: src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py [hunk 2]
```diff
@@ -26,25 +26,33 @@ def tf_type_to_ov_type(tf_type_int):
...
+        tf_shape = attr.shape
+        if tf_shape.unknown_rank:
+            return PartialShape.dynamic()
+        shape_dims = tf_shape.dim
+        shape = [dim.size for dim in shape_dims]
+        return PartialShape(shape)
+    elif attr_type is None:
```
Comment (ilya-lavrenov) [inline src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py:55]: can we add tests for this logic?

Comment (rkazants) [reply src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py:55]: https://www.kaggle.com/models/google/mobile-object-localizer-v1/frameworks/tensorFlow1/variations/object-detection-mobile-object-localizer-v1/versions/1 covers this logic. Without this fix, it fails due to this bug. Now it fails due to inference results mismatch that is the next step to fix.

**Review by popovaan — APPROVED** [review]

**Pull request merged**